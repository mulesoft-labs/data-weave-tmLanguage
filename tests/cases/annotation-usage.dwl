/**
* Annotation that marks a variable declaration for lazy initialization.
*/
@AnnotationTarget(targets = ["Variable"])
@Since(version = "2.3.0")
annotation Lazy()