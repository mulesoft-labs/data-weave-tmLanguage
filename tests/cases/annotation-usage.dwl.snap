>/**
#^^ source.data-weave comment.block.dw punctuation.definition.comment.dw
#  ^^ source.data-weave comment.block.dw
>* Annotation that marks a variable declaration for lazy initialization.
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.data-weave comment.block.dw
>*/
#^^ source.data-weave comment.block.dw punctuation.definition.comment.dw
>@AnnotationTarget(targets = ["Variable"])
#^ source.data-weave meta.annot.usage.dw storage.type.dw
# ^^^^^^^^^^^^^^^^ source.data-weave meta.annot.usage.dw entity.name.function.dw
#                 ^ source.data-weave meta.annot.usage.dw
#                  ^^^^^^^ source.data-weave meta.annot.usage.dw variable.parameter.dw
#                         ^ source.data-weave meta.annot.usage.dw
#                          ^ source.data-weave meta.annot.usage.dw keyword.operator.declaration.dw
#                           ^ source.data-weave meta.annot.usage.dw
#                            ^ source.data-weave meta.annot.usage.dw meta.array.literal.dw meta.brace.square.dw
#                             ^ source.data-weave meta.annot.usage.dw meta.array.literal.dw string.quoted.double.dw punctuation.definition.string.begin.dw
#                              ^ source.data-weave meta.annot.usage.dw meta.array.literal.dw string.quoted.double.dw
#                               ^ source.data-weave meta.annot.usage.dw meta.array.literal.dw string.quoted.double.dw
#                                ^ source.data-weave meta.annot.usage.dw meta.array.literal.dw string.quoted.double.dw
#                                 ^ source.data-weave meta.annot.usage.dw meta.array.literal.dw string.quoted.double.dw
#                                  ^ source.data-weave meta.annot.usage.dw meta.array.literal.dw string.quoted.double.dw
#                                   ^ source.data-weave meta.annot.usage.dw meta.array.literal.dw string.quoted.double.dw
#                                    ^ source.data-weave meta.annot.usage.dw meta.array.literal.dw string.quoted.double.dw
#                                     ^ source.data-weave meta.annot.usage.dw meta.array.literal.dw string.quoted.double.dw
#                                      ^ source.data-weave meta.annot.usage.dw meta.array.literal.dw string.quoted.double.dw punctuation.definition.string.end.dw
#                                       ^ source.data-weave meta.annot.usage.dw meta.array.literal.dw meta.brace.square.dw
#                                        ^ source.data-weave meta.annot.usage.dw
>@Since(version = "2.3.0")
#^ source.data-weave meta.annot.usage.dw storage.type.dw
# ^^^^^ source.data-weave meta.annot.usage.dw entity.name.function.dw
#      ^ source.data-weave meta.annot.usage.dw
#       ^^^^^^^ source.data-weave meta.annot.usage.dw variable.parameter.dw
#              ^ source.data-weave meta.annot.usage.dw
#               ^ source.data-weave meta.annot.usage.dw keyword.operator.declaration.dw
#                ^ source.data-weave meta.annot.usage.dw
#                 ^ source.data-weave meta.annot.usage.dw string.quoted.double.dw punctuation.definition.string.begin.dw
#                  ^ source.data-weave meta.annot.usage.dw string.quoted.double.dw
#                   ^ source.data-weave meta.annot.usage.dw string.quoted.double.dw
#                    ^ source.data-weave meta.annot.usage.dw string.quoted.double.dw
#                     ^ source.data-weave meta.annot.usage.dw string.quoted.double.dw
#                      ^ source.data-weave meta.annot.usage.dw string.quoted.double.dw
#                       ^ source.data-weave meta.annot.usage.dw string.quoted.double.dw punctuation.definition.string.end.dw
#                        ^ source.data-weave meta.annot.usage.dw
>annotation Lazy()
#^^^^^^^^^^ source.data-weave meta.directive.annot.dw storage.type.dw
#          ^ source.data-weave meta.directive.annot.dw
#           ^^^^ source.data-weave meta.directive.annot.dw entity.name.function.dw
#               ^ source.data-weave meta.directive.annot.dw
#                ^ source.data-weave meta.directive.annot.dw